File: starters/tken-python-openai-compatible-cookbook/examples/retry_policy.py
import time


def retry(operation, attempts=3, delay=0.5):
    last_error = None
    for attempt in range(1, attempts + 1):
        try:
            return operation()
        except Exception as exc:
            last_error = exc
            if attempt < attempts:
                time.sleep(delay * attempt)
    raise last_error


if __name__ == "__main__":
    print({"ok": True, "policy": "linear backoff", "attempts": 3})

File: starters/tken-node-openai-compatible-cookbook/examples/retry-policy.mjs
export async function retry(operation, attempts = 3) {
  let lastError;
  for (let attempt = 1; attempt <= attempts; attempt += 1) {
    try {
      return await operation();
    } catch (error) {
      lastError = error;
      if (attempt < attempts) await new Promise((resolve) => setTimeout(resolve, attempt * 250));
    }
  }
  throw lastError;
}

console.log(JSON.stringify({ ok: true, policy: "linear backoff", attempts: 3 }));
